### DISCUSSION TITLE
How do I execute an Action on a transition?

### DISCUSSION THREAD

**Discussion by arscroggin** [dotnet-state-machine/stateless#601] [Q&A]
How do I execute an action on a transition from state A to state B? Something like this:

`stateMachine.Configure(State.A)`
`    .Permit(Trigger.MyTrigger, State.B, DoAction()`

It seems though you can only specify a Trigger and the Destination State.

Am I missing something? Thanks in advance!

**Comment by ThisIsNoZaku** [comment]
Use OnEntry on the destination state or OnExit on the origin.
